### PR TITLE
Create User context

### PR DESCRIPTION
**Pull request by bryanparmelee**
- Updated login page to use newly added signInAuthUserWithEmailAndPassword function from api/firebase
- Created UserContext.tsx , which uses the firebase onAuthStateChanged listener to track logins and sign outs and update the current user accordingly. 
- Add the UserContext provider to layout.tsx to make the current user context available to any component that needs it via the useContext hook. 
- Imported the userContext into the Header component so the sign out option appears in the burger menu instead of login after a user is signed in.
 - Verified that signing in and out works using one of our test users.

### CODE REVIEW

File: src/contexts/UserContext.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+export const UserProvider = ({ children }: Props) => {
+    const [currentUser, setCurrentUser] = useState<User | null>(null)
+    const value = {
+        currentUser,
+        setCurrentUser
+    }
+
+    useEffect(() => {
+        const unsubscribe = onAuthStateChangedListener((user) => {
```
Comment (NathanWEdwards) [inline src/contexts/UserContext.tsx:29]: `unsubscribe` is always undefined? _onAuthStateChangedListener()_ has no return statement.

**Review by NathanWEdwards — CHANGES_REQUESTED** [review]
This request includes style changes. Including semicolons removes ambiguities that arise from lines of code's intended use and what is actually executed considering automatic semicolon insertion.

An issue has been added to address the react-jsx-in-scope linter error.

I haven't finished reviewing. I did notice that onAuthStateChangedListener() lacks a return statement. It's usage in the User context indicates it should.

File: src/contexts/UserContext.tsx (hunk 1 above)
Comment (bryanparmelee) [reply src/contexts/UserContext.tsx]: Ah yes, good call. onAuthStateChangedListener() returns the firebase.Unsubscribe interface that is returned by firebase's onAuthStateChanged method. I added this type to onAuthStateChangedListener, as well as a return, so the unsubscribe inside of the useEffect is now also typed.

File: src/contexts/UserContext.tsx [hunk 2]
```diff
@@ -25,14 +25,10 @@ export const UserProvider = ({ children }: Props) => {
         setCurrentUser
     }
 
+    console.log(currentUser)
```
Comment (NathanWEdwards) [inline src/contexts/UserContext.tsx:28]: Should we exclude console.log() statements on the main branch?

**Review by NathanWEdwards — APPROVED** [review]

File: src/contexts/UserContext.tsx (hunk 2 above)
Comment (bryanparmelee) [reply src/contexts/UserContext.tsx]: Yea, definitely. Totally forgot to remove that after I was done making edits.

**Pull request merged**